backend/common.go

```diff
@@ -91,6 +91,16 @@ func (s *DataSourceInstanceSettings) HTTPClientOptions() (httpclient.Options, er
 		return httpclient.Options{}, err
 	}
 
+	if s.BasicAuthEnabled {
```

Comment:
i checked parseHTTPSettings, it seems the basicauth is parsed into httpSettings, here we would like to overwrite it, did i missed something.

Reply:
saw your comments